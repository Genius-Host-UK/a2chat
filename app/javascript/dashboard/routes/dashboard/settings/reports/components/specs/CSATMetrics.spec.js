import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import CsatMetrics from '../CsatMetrics.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('CsatMetrics.vue', () => {
  let getters;
  let store;

  beforeEach(() => {
    getters = {
      'csat/getMetrics': () => ({ totalResponseCount: 100 }),
      'csat/getRatingPercentage': () => ({ 1: 10, 2: 20, 3: 30, 4: 30, 5: 10 }),
      'csat/getSatisfactionScore': () => 85,
      'csat/getResponseRate': () => 90,
    };

    store = new Vuex.Store({
      getters,
    });
  });

  it('computes response count correctly', () => {
    const wrapper = shallowMount(CsatMetrics, {
      store,
      localVue,
      mocks: {
        $t: msg => msg,
      },
      stubs: ['csat-metric-card', 'woot-horizontal-bar'],
    });

    expect(wrapper.vm.responseCount).toBe('100');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('formats values to percent correctly', () => {
    const wrapper = shallowMount(CsatMetrics, {
      store,
      localVue,
      mocks: {
        $t: msg => msg,
      },
      stubs: ['csat-metric-card', 'woot-horizontal-bar'],
    });

    expect(wrapper.vm.formatToPercent(85)).toBe('85%');
    expect(wrapper.vm.formatToPercent(null)).toBe('--');
  });

  it('maps rating value to emoji correctly', () => {
    const wrapper = shallowMount(CsatMetrics, {
      store,
      localVue,
      mocks: {
        $t: msg => msg,
      },
      stubs: ['csat-metric-card', 'woot-horizontal-bar'],
    });

    const rating = wrapper.vm.csatRatings[0]; // assuming this is { value: 1, emoji: 'ðŸ˜¡' }
    expect(wrapper.vm.ratingToEmoji(rating.value)).toBe(rating.emoji);
  });
});
