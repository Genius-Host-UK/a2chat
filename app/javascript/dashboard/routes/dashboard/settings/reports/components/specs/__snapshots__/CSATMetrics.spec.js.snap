// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsatMetrics.vue computes response count correctly 1`] = `
<div class="row csat--metrics-container">
  <csat-metric-card-stub label="CSAT_REPORTS.METRIC.TOTAL_RESPONSES.LABEL" value="100" infotext="CSAT_REPORTS.METRIC.TOTAL_RESPONSES.TOOLTIP"></csat-metric-card-stub>
  <csat-metric-card-stub label="CSAT_REPORTS.METRIC.SATISFACTION_SCORE.LABEL" value="85%" infotext="CSAT_REPORTS.METRIC.SATISFACTION_SCORE.TOOLTIP"></csat-metric-card-stub>
  <csat-metric-card-stub label="CSAT_REPORTS.METRIC.RESPONSE_RATE.LABEL" value="90%" infotext="CSAT_REPORTS.METRIC.RESPONSE_RATE.TOOLTIP"></csat-metric-card-stub>
  <div class="medium-6 report-card">
    <h3 class="heading">
      <div class="emoji--distribution">
        <div class="emoji--distribution-item"><span class="emoji--distribution-key">😞</span> <span>10%</span></div>
        <div class="emoji--distribution-item"><span class="emoji--distribution-key">😑</span> <span>20%</span></div>
        <div class="emoji--distribution-item"><span class="emoji--distribution-key">😐</span> <span>30%</span></div>
        <div class="emoji--distribution-item"><span class="emoji--distribution-key">😀</span> <span>30%</span></div>
        <div class="emoji--distribution-item"><span class="emoji--distribution-key">😍</span> <span>10%</span></div>
      </div>
    </h3>
    <div class="emoji--distribution-chart">
      <woot-horizontal-bar-stub collection="[object Object]" height="24"></woot-horizontal-bar-stub>
    </div>
  </div>
</div>
`;
